src/SLIP10Node.ts

```diff
@@ -34,142 +37,209 @@ export type JsonSLIP10Node = {
 };
 
 export type SLIP10NodeInterface = JsonSLIP10Node & {
+  chainCodeBuffer: Buffer;
+
   /**
-   * The raw bytes of the key material for this node, as a Node.js Buffer or
-   * browser-equivalent.
+   * The private key for this node, as a Node.js Buffer or browser-equivalent.
+   * May be undefined if this node is a public key only node.
    */
-  keyBuffer: Buffer;
+  privateKeyBuffer?: Buffer;
+
+  /**
+   * The public key for this node, as a Node.js Buffer or browser-equivalent.
+   */
+  publicKeyBuffer: Buffer;
 
   /**
    * @returns A JSON-compatible representation of this node's data fields.
    */
   toJSON(): JsonSLIP10Node;
 };
 
-type SLIP10NodeKeyDepthOptions = {
+type SLIP10NodeConstructorOptions = {
   readonly depth: number;
-  readonly key: Buffer | string;
-  readonly derivationPath?: never;
+  readonly chainCode: Buffer;
+  readonly privateKey?: Buffer;
+  readonly publicKey: Buffer;
   readonly curve: SupportedCurve;
 };
 
-type SLIP10NodeDerivationPathOptions = {
-  readonly derivationPath: RootedSLIP10PathTuple;
-  readonly depth?: never;
-  readonly key?: never;
+type SLIP10ExtendedKeyOptions = {
+  readonly depth: number;
+  readonly chainCode: Buffer;
+  readonly privateKey?: Buffer;
+  readonly publicKey?: Buffer;
   readonly curve: SupportedCurve;
 };
 
-export type SLIP10NodeOptions =
-  | SLIP10NodeKeyDepthOptions
-  | SLIP10NodeDerivationPathOptions;
-
-type SLIP10NodeConstructorOptions = {
-  readonly depth: number;
-  readonly key: Buffer;
+type SLIP10DerivationPathOptions = {
+  readonly derivationPath: RootedSLIP10PathTuple;
   readonly curve: SupportedCurve;
 };
 
 export class SLIP10Node implements SLIP10NodeInterface {
```

Comment:
Is it intentional to not be able to create a `SLIP10Node` from a JSON node? Same goes for BIP44